### ISSUE TITLE
flyway import: "duplicate key" error with exactly one row in one table

### ISSUE THREAD

**Issue by 2019-05-10** [zonkyio/embedded-database-spring-test#141]
Trying out flyway migration/data import I am running into what seems to be a bug:
a) the DDL has exactly one table
b) the data has exactly one row of data for that table
and yet I get
```
2020-12-11 11:55:15.904  INFO 1194490 --- [es:pid(1194628)] i.z.t.d.p.embedded.EmbeddedPostgres      : 2020-12-11 11:55:15.904 CET [1194674] ERROR:  duplicate key value violates unique constraint "foo_pkey"
2020-12-11 11:55:15.904  INFO 1194490 --- [es:pid(1194628)] i.z.t.d.p.embedded.EmbeddedPostgres      : 2020-12-11 11:55:15.904 CET [1194674] DETAIL:  Key (prefix, id)=(MYPREFIX, 00003444) already exists.
2020-12-11 11:55:15.904  INFO 1194490 --- [es:pid(1194628)] i.z.t.d.p.embedded.EmbeddedPostgres      : 2020-12-11 11:55:15.904 CET [1194674] CONTEXT:  COPY foo, line 1
2020-12-11 11:55:15.904  INFO 1194490 --- [es:pid(1194628)] i.z.t.d.p.embedded.EmbeddedPostgres      : 2020-12-11 11:55:15.904 CET [1194674] STATEMENT:  COPY public.foo (prefix, id, longnumber) FROM stdin
```
which from my PoV is patently nonsense since there's exactly one record and no more, hence the key cannot possibly exist more than once.

Attached V1.0__.sql and afterMigrate.sql
[duplicate_key_with_single_record.zip](https://github.com/zonkyio/embedded-database-spring-test/files/5678779/duplicate_key_with_single_record.zip)

**Comment by 2019-05-10** [comment]
Once flyway is enabled, SpringBoot triggers the migration as well, causing the issue.
There is apaprently no way to disable flyway for SpringBoot, but leave it enabled for embbeded-db-spring-test.

Following [https://www.baeldung.com/database-migrations-with-flyway](https://www.baeldung.com/database-migrations-with-flyway) I had to create custom FlywayMigrationStrategy bean that does nothing, sidestepping SpringBoot:

```
import org.springframework.boot.autoconfigure.flyway.FlywayMigrationStrategy;
import org.springframework.context.annotation.*;

//https://www.baeldung.com/database-migrations-with-flyway
@Configuration
public class TestFlywayMigrationStrategyConfig {
	@Bean
	public FlywayMigrationStrategy flywayMigrationStrategy() {
		return flyway -> {
			// do nothing
		};
	}
}
```

Maybe it would be a good idea to include such a bean?

**Comment by tomix26** [comment]
It seems that you prepare the data in `afterMigrate.sql` callback. At this moment, it's not guaranteed that the migration run and therefore the after migrate callback will be called only once. It's a limitation of the current design of the library.

So either you have to adapt the callback migration to be able to be applied multiple times, or you can try to use the current beta version [2.0.0-beta1](https://github.com/zonkyio/embedded-database-spring-test/releases/tag/v2.0.0-beta1), which internally uses AOP techniques and this problem should already be fixed.

**Comment by tomix26** [comment]
No response, I'm closing the issue. I think it should be solved in 2.0.0, if not, feel free to reopen the issue.
